tests/tracer/test_logging.py

```diff
@@ -0,0 +1,356 @@
+    """
+    When using ddtrace-run with a custom logger,
+        the ddtrace logger does not override any custom logs settings.
+    """
+    env = os.environ.copy()
+    env["DD_TRACE_DEBUG"] = "true"
+    log_file = tmpdir + "/testlog.log"
+    env["DD_TRACE_LOG_FILE"] = log_file
+    env["DD_TRACE_FILE_SIZE_BYTES"] = "200000"
+    code = """
+import logging
+custom_logger = logging.getLogger('custom')
+custom_logger.setLevel(logging.WARN)
+
+assert custom_logger.parent.name == 'root'
+assert custom_logger.level == logging.WARN
+"""
+
+    out, err, status, pid = ddtrace_run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert err == b""
+    assert out == b""
+    with open(log_file) as file:
+        content = file.read()
+        assert len(content) > 0
+        assert re.search(LOG_PATTERN, content) is not None
+
+
+def test_unrelated_logger_in_warn_with_ddtrace_run(ddtrace_run_python_code_in_subprocess):
+    """
+    When using ddtrace-run and a custom logger without debug mode enabled,
+        the ddtrace logger does not override any custom logs settings.
+    """
+    env = os.environ.copy()
+    env["DD_TRACE_DEBUG"] = "false"
+    code = """
+import logging
+custom_logger = logging.getLogger('custom')
+custom_logger.setLevel(logging.WARN)
+
+assert custom_logger.parent.name == 'root'
+assert custom_logger.level == logging.WARN
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.level == logging.WARN
+assert len(ddtrace_logger.handlers) == 1
+assert isinstance(ddtrace_logger.handlers[0], logging.StreamHandler)
+
+ddtrace_logger.warning('warning log')
+"""
+
+    out, err, status, pid = ddtrace_run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert re.search(LOG_PATTERN, str(err)) is not None
+    assert b"warning log" in err
+    assert out == b""
+
+
+def test_warn_logs_go_to_stderr_default(run_python_code_in_subprocess, ddtrace_run_python_code_in_subprocess):
+    """
+    When setting up the default ddtrace logger when not in debug mode,
+        it automatically logs to stderr at WARN log level.
+    """
+    code = """
+import logging
+import ddtrace
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.level == logging.WARN
+assert len(ddtrace_logger.handlers) == 1
+assert isinstance(ddtrace_logger.handlers[0], logging.StreamHandler)
+
+ddtrace_logger.warning('warning log')
+"""
+
+    out, err, status, pid = run_python_code_in_subprocess(code)
+    assert status == 0, err
+    assert re.search(LOG_PATTERN, str(err)) is not None
+    assert b"warning log" in err
+    assert out == b""
+
+    code = """
+import logging
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.level == logging.WARN
+assert len(ddtrace_logger.handlers) == 1
+assert isinstance(ddtrace_logger.handlers[0], logging.StreamHandler)
+
+ddtrace_logger.warning('warning log')
+"""
```

Comment:
```suggestion
```

This is already defined, no reason to re-define it.

Reply:
There is one small difference in that the `patch` version imports in `ddtrace`

```
import logging
import ddtrace
```

While `ddtrace-run` doesn't, so there's only this import:

```
import logging
```

. Does this difference matter for testing? 

If `ddtrace-run` can be assumed to work  the same way when `import ddtrace` is used, then I can definitely get rid the duplicate `code`.

Reply:
oh! you are right... they are different 🤦🏻